### PR TITLE
fix(core): fail to clean up framework observers

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

Added logic to clean up the Framework detection observers, and ignore new incoming observers after the detection completes.

#### Issue #, if available

https://github.com/aws-amplify/amplify-js/issues/11770

#### Description of how you validated changes

Unit tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by erinleigh90 — APPROVED** [review]
LGTM

**Review by erinleigh90 — DISMISSED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by cwomack — DISMISSED** [review]
LGTM!

**Review by jimblanc — DISMISSED** [review]

**Review by erinleigh90 — APPROVED** [review]

File: packages/core/__tests__/Platform-test.ts [hunk 1]
```diff
@@ -81,3 +81,29 @@ describe('Platform test', () => {
...
 });
+
+describe('detectFramework observers', () => {
+	let module;
+
+	beforeAll(() => {
+		jest.resetModules();
+		module = require('../src/Platform/detectFramework');
+	});
+
```
Comment (david-mcafee) [inline packages/core/__tests__/Platform-test.ts:92]: Your test uses fake timers, but they are never reset. Would suggest cleaning that up in an `afterEach` block, maybe something like: 

```suggestion

afterEach(() => {
    jest.useRealTimers();
});
```

**Review by david-mcafee — CHANGES_REQUESTED** [review]

File: packages/core/__tests__/Platform-test.ts [hunk 2]
```diff
@@ -9,7 +9,18 @@ import { detectFramework, clearCache } from '../src/Platform/detectFramework';
 import * as detection from '../src/Platform/detection';
 
 describe('Platform test', () => {
-	beforeEach(() => clearCache());
+	beforeAll(() => {
+		jest.useFakeTimers();
+	});
+
+	afterAll(() => {
+		jest.useRealTimers();
```
Comment (david-mcafee) [inline packages/core/__tests__/Platform-test.ts:17]: nit: I'd prefer using `afterEach` (see [previous suggestion](https://github.com/aws-amplify/amplify-js/pull/11782/files#r1291633571)) for better isolation between tests, but not a blocker.

**Review by david-mcafee — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**